internal/oci/options.go

```diff
@@ -0,0 +1,61 @@
+package oci
+
+import (
+	"fmt"
+
+	"go.flipt.io/flipt/internal/containers"
+	"go.flipt.io/flipt/internal/oci/ecr"
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+// StoreOptions are used to configure call to NewStore
+// This shouldn't be handled directory, instead use one of the function options
+// e.g. WithBundleDir or WithCredentials
+type StoreOptions struct {
+	bundleDir       string
+	manifestVersion oras.PackManifestVersion
+	auth            credentialFunc
+}
+
+// WithCredentials configures username and password credentials used for authenticating
+// with remote registries
+func WithCredentials(kind, user, pass string) (containers.Option[StoreOptions], error) {
+	switch kind {
+	case "aws-ecr":
+		return WithAWSECRCredentials(), nil
+	case "static", "":
```

Comment:
As we're setting the default value to be `static` we don't need tp consider the empty thing!

Reply:
I wish that there will be a simple way to set OCIAuthentication.Type to `static` as a default value.

Reply:
But there's, [StorageConfig.setDefaults](https://github.com/flipt-io/flipt/blob/main/internal/config/storage.go#L72)!